core/project_settings.h

```diff
@@ -66,6 +66,14 @@ class ProjectSettings : public Object {
 
 	int _dirty_this_frame = 2;
 
+	// Starting version from 1 ensures that all callers can reset their tested version to 0,
+	// and will always detect the initial project settings as a "change".
+	uint32_t _version = 1;
+
+	// For our GET_GLOBAL macro we store static cached values,
+	// and we want to prevent any race conditions for them, so provide a mutex.
```

Comment:
Mutexes provide substantial overhead. We should make sure it's still noticeably faster than the original `GLOBAL_GET` was.

Reply:
Have added timings in the notes for the PR, is quite a bit faster as expected.